rpmio/macro.c

```diff
@@ -1174,6 +1177,89 @@ static void doShescape(MacroBuf mb, rpmMacroEntry me, ARGV_t argv, size_t *parse
     mbAppend(mb, '\'');
 }
 
+static unsigned long getmem_total(void)
+{
+    unsigned long mem = 0;
+    long int pagesize = sysconf(_SC_PAGESIZE);
+    long int pages = sysconf(_SC_PHYS_PAGES);
+
+    if (pagesize < 0)
+	pagesize = 4096;
+    if (pages > 0)
+	mem = pages * pagesize;
+
+    return mem;
+}
+
+static unsigned long getmem_proc(int thread)
```

Comment:
Wouldn't it be better to add a separate `getmem_thread()` function for the thread case? There doesn't seem to be much they have in common, apart from the `getmem_total()` call.